alfresco-googledrive-repo-base/src/main/java/org/alfresco/integrations/google/docs/webscripts/UploadContent.java

```diff
@@ -48,195 +54,188 @@
+                            googledocsService.addRemotePermissions(credential,
+                                file,
+                                googledocsService.getGooglePermissions(nodeRef,
+                                    PROP_CURRENT_PERMISSIONS)
+                            );
+                        }
+                    }
                 }
-                catch(GoogleDocsServiceException gdse)
+                else
                 {
                     file = googledocsService.uploadFile(credential, nodeRef);
                     if (log.isDebugEnabled())
                     {
                         log.debug(nodeRef + " Uploaded to Google.");
                     }
-                    // Re-apply the previous permissions, if they exist
-                    if (nodeService.getProperty(nodeRef, GoogleDocsModel.PROP_CURRENT_PERMISSIONS) != null)
+                }
+
+                if (jsonParams.containsKey(PARAM_PERMISSIONS))
+                {
+                    if (log.isDebugEnabled())
                     {
-                        googledocsService.addRemotePermissions(credential,
-                                file,
-                                googledocsService.getGooglePermissions(nodeRef, GoogleDocsModel.PROP_CURRENT_PERMISSIONS)
-                        );
+                        log.debug("Adding permissions to remote object");
                     }
+                    googledocsService.addRemotePermissions(credential,
+                        file,
+                        (List<GooglePermission>) jsonParams.get(PARAM_PERMISSIONS)
+                    );
                 }
-            }
-            else
-            {
-                file = googledocsService.uploadFile(credential, nodeRef);
-                if (log.isDebugEnabled())
+
+                // If this is a non-cloud instance of Alfresco, we need to make the
+                // node versionable before we start working on it. We want the the
+                // version component to be triggered on save. The versionable aspect
+                // is only added if this is existing content, not if it was just
+                // created where the document is the initial version when saved
+                if (!nodeService.hasAspect(nodeRef, ASPECT_TEMPORARY)
+                    && !nodeService.hasAspect(nodeRef, ASPECT_VERSIONABLE))
                 {
-                    log.debug(nodeRef + " Uploaded to Google.");
+                    Map<String, Serializable> versionProperties = new HashMap<>();
+                    versionProperties.put(Version2Model.PROP_VERSION_TYPE, VersionType.MAJOR);
+
+                    nodeService.setProperty(nodeRef, PROP_AUTO_VERSION, true);
+                    // autoVersionOnUpdateProps now set to false to follow Share upload scripts (fixes GOOGLEDOCS-111)
+                    nodeService.setProperty(nodeRef, PROP_AUTO_VERSION_PROPS, false);
+
+                    log.debug(
```

Comment:
Trivial: for consistency, wrap in an if block (as per rest of code) ?

Reply:
This is why we should really use SLF4J, so we don't need all these verbose checks. But that's a change for another time.
Done.